### ISSUE TITLE
Upgrading on Linux

### ISSUE THREAD

**Issue by toxpal** [facelessuser/Rummage#261]
Maybe it's a lame question, but shouldn't update remove old version and replace it with a new one?

Today I saw v4.2 is available, so ran _pip install --upgrade rummage_ command, which succeeded. Still, trying to run Rummage opens 4.1.3 for me. I thought that maybe a reboot is needed, but after reboot it's the same:

`python -m rummage`

once again opens 4.1.3. Tried to execute _pip install --upgrade rummage_ again, but got this message:

`Requirement already up-to-date: rummage in /usr/lib/python3.6/site-packages (4.2)`

What do I miss here? I see _rummage-4.2-py3.6.egg-info_ directory in _/usr/lib/python3.6/site-packages_, but I also see this - _/home/MY_USERNAME/.local/lib/python3.6/site-packages/rummage-4.1.3-py3.6.egg-info/_

In short, /usr only contains 4.2, and /home only contains 4.1.3, and python command executes 4.1.3 by default.

**Comment by toxpal** [comment]
P.S. Somehow, after upgrade, I can now launch Rummage by just typing _rummage_ into terminal, and it opens 4.2 by default. Previously, I couldn't even execute _rummage_ command on my system, the only way was _python -m rummage_ command.

**Comment by facelessuser** [comment]
> Maybe it's a lame question, but shouldn't update remove old version and replace it with a new one?

Yes, and it does.

> What do I miss here?

I'm not sure.  I honestly don't have enough information about your system to even really guess.

I know Linux is a bit different than macOS and Linux is some ways.  I know Linux will install things in different places if you run `sudo pip install` vs `pip install` (global vs local).

If you have multiple Python versions, sometimes you can get confused which version you are running.  I have a number of Python versions on my macOS system as that is where I do *most* of my Python development. So sometimes this happens to me. `python` for me is Python 2. `python3` is usually the latest and greatest, but not always the Python version I'm currently defaulting to.

So I often have to be more explicit and run either `python3.6 -m rummage` or `rummage3.6` (pip *should* install `rummage` and `rummageX.X` where `X.X` is the corresponding major and minor version number of the Python it is installed under). Calling rummage should be available assuming install went proper and created the bash script in your Python bin/script folder, and that bin/script folder is in the Path (directories that Bash is instructed to look first to find an executable when an executable is given with no path).

If rummage is running while you do `pip install --upgrade`, sometimes everything doesn't get upgraded proper, so keep in mind you need to shut down rummage before upgrading.

There are a lot of things that could have happened.  I don't know enough about your system to really guess what happened.  Assuming that calling `rummage` is using the same Python version that calling `python -m rummage` uses, the Rummage version **should** be the same.  *Maybe* this can have alternate versions if one is referencing a local install vs a global install (sudo vs no sudo)?

If you want, you can always `pip uninstall rummage` and `sudo pip uninstall rummage` and test to make sure you can't run rummage at all no matter what way you run it.  Then try and install it again.

Again, when having multiple python versions, `pip` can reference a specific Python version as well. So I'm usually explicit in how I run pip as well. `pip3` or `pip3.6` to be really specific.  Sometimes I'll even run it as `python3.6 -m pip install` as that will call the Python version 3.6 and then use the pip module associated with that version.

I packed a lot of info in this response...hopefully something in here is useful.

**Comment by toxpal** [comment]
Thanks, will try with sudo... Anyhow, the most important thing is that Rummage now displays proper colors in dark themes and has other serious improvements.

**Comment by facelessuser** [comment]
Glad it's working better for you.  There will most likely be a bug fix release later today or tomorrow to address focusing issues in Linux.
